libusb/hid.c

```diff
@@ -909,12 +922,17 @@ static void *read_thread(void *param)
 }
 
 
-static int hidapi_initialize_device(hid_device *dev, const struct libusb_interface_descriptor *intf_desc)
+static int hidapi_initialize_device(hid_device *dev, char* path, const struct libusb_interface_descriptor *intf_desc)
 {
 	int i =0;
 	int res = 0;
 	struct libusb_device_descriptor desc;
-	libusb_get_device_descriptor(libusb_get_device(dev->device_handle), &desc);
+	libusb_get_device_descriptor(libusb_get_device(dev->device_handle), &desc);	
+
+	dev->device_info = (struct hid_device_info*) calloc(1, sizeof(struct hid_device_info));
+	dev->device_info->next = NULL;
+	dev->device_info->path = path;
```

Comment:
I don't like that we're passing the ownership inside here
lets initialize this field at the same place, where we call `hidapi_initialize_device` - where the ownership of `path` is hold